tests/cli/test_cli_get_sample.py

```diff
@@ -0,0 +1,308 @@
+
+
+def test_get_sample_bad_sample(invoke_cli, disk_store: Store):
+    """Test to get a sample using a non-existing sample-id """
+    # GIVEN an empty database
+
+    # WHEN getting a sample
+    db_uri = disk_store.uri
+    name = 'dummy_name'
+    result = invoke_cli(['--database', db_uri, 'get', 'sample', name])
+
+    # THEN then it should complain on missing sample id instead of getting a sample
+    assert result.exit_code == 0
+
+
+def test_get_sample_required(invoke_cli, disk_store: Store):
+    """Test to get a sample using only the required arguments"""
+    # GIVEN a database with a sample
+    sample_id = add_sample(disk_store).internal_id
+    assert disk_store.Sample.query.count() == 1
+
+    # WHEN getting a sample
+    db_uri = disk_store.uri
+
+    result = invoke_cli(
+        ['--database', db_uri, 'get', 'sample', sample_id])
+
+    # THEN then it should have been get
+    assert result.exit_code == 0
+
+
+def test_get_samples_required(invoke_cli, disk_store: Store):
+    """Test to get several samples using only the required arguments"""
+    # GIVEN a database with two samples
+    sample_id1 = add_sample(disk_store, '1').internal_id
+    sample_id2 = add_sample(disk_store, '2').internal_id
+    assert disk_store.Sample.query.count() == 2
+
+    # WHEN getting a sample
+    db_uri = disk_store.uri
+
+    result = invoke_cli(
+        ['--database', db_uri, 'get', 'sample', sample_id1, sample_id2])
+
+    # THEN then it should have been get
+    assert result.exit_code == 0
+
+
+def test_get_sample_output(invoke_cli, disk_store: Store):
+    """Test that the output has the data of the sample"""
+    # GIVEN a database with a sample with data
+    sample = add_sample(disk_store)
+    sample_id = sample.internal_id
+    name = sample.name
+    customer_id = sample.customer.internal_id
+    application_tag = sample.application_version.application.tag
+    state = sample.state
+    priority_human = sample.priority_human
+
+    # WHEN getting a sample
+    db_uri = disk_store.uri
+
+    result = invoke_cli(
+        ['--database', db_uri, 'get', 'sample', sample_id])
+
+    # THEN then it should have been get
+    assert result.exit_code == 0
+    assert sample_id in result.output
+    assert name in result.output
+    assert customer_id in result.output
+    assert application_tag in result.output
+    assert state in result.output
+    assert priority_human in result.output
+
+
+def test_get_sample_external_false(invoke_cli, disk_store: Store):
+    """Test that the output has the data of the sample"""
+    # GIVEN a database with a sample with data
+    sample_id = add_sample(disk_store, is_external=False).internal_id
+    is_external = 'No'
+
+    # WHEN getting a sample
+    db_uri = disk_store.uri
+
+    result = invoke_cli(
+        ['--database', db_uri, 'get', 'sample', sample_id])
+
+    # THEN then it should have been get
+    assert result.exit_code == 0
+    assert is_external in result.output
```

Comment:
Let's hope that's the onle 'No' in the output then ;)

Reply:
Now we test for both 'Yes' and 'No' for both is_external=true/false, ok?

Reply:
Fine by me!